src/experiments/73.collision-lightning.tsx

```diff
@@ -0,0 +1,269 @@
+import {
+  Environment,
+  MeshReflectorMaterial,
+  PerspectiveCamera,
+  Text3D
+} from '@react-three/drei'
+import { useFrame } from '@react-three/fiber'
+import {
+  BallCollider,
+  ContactForcePayload,
+  Physics,
+  RigidBody
+} from '@react-three/rapier'
+import { folder, useControls } from 'leva'
+import { gsap } from 'lib/gsap/index'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import * as THREE from 'three'
+
+import { R3FSuspenseLayout } from '~/components/layout/r3f-suspense-layout'
+import { useDeviceDetect } from '~/hooks/use-device-detect'
+
+type TConfig = {
+  emissive: string
+  text: string
+  minForce: number
+  forceIntensity: number
+  transmissionSampler: boolean
+  backside: boolean
+  samples: number
+  resolution: number
+  transmission: number
+  roughness: number
+  thickness: number
+  ior: number
+  chromaticAberration: number
+  anisotropy: number
+  distortion: number
+  distortionScale: number
+  temporalDistortion: number
+  clearcoat: number
+  attenuationDistance: number
+  attenuationColor: string
+  color: string
+}
+
+interface GlyphProps {
+  letter: string
+  config: TConfig
+  isMobile: boolean | undefined
+}
+
+const Glyph = ({ letter, config, isMobile }: GlyphProps) => {
+  const r = THREE.MathUtils.randFloatSpread
+  const api = useRef<any>(null)
+  const pos = useMemo(() => new THREE.Vector3(r(10), r(10), r(10)), [r])
+  const rot = useMemo(() => new THREE.Euler(r(10), r(10), r(10)), [r])
```

Comment:
lets move this const r outside of the component so we can remove the dependency for pos and rot